search/tasks.py

```diff
@@ -19,25 +19,41 @@
 from search.models import PercolateQuery
 
 
+def lookup_id(obj_or_id):
+    """
+    Handle the current and deprecated paths while the Celery queue may possibly have old and new tasks queued up.
+    """
+    if isinstance(obj_or_id, int):
+        return obj_or_id
+    else:
+        # Deprecated!
+        return obj_or_id.id
+
+
 @async.task
-def remove_program_enrolled_user(user):
+def remove_program_enrolled_user(program_enrollment_id):
     """
     Remove program-enrolled user from index
 
     Args:
-        user (User): A program-enrolled user to remove from index
+        program_enrollment_id (int): A ProgramEnrollment to remove from the index
     """
-    _remove_program_enrolled_user(user)
+    # Deprecation warning!
+    program_enrollment_id = lookup_id(program_enrollment_id)
+    _remove_program_enrolled_user(program_enrollment_id)
```

Comment:
I'm only using it when the tasks are being dispatched, so in the calling functions to here.